sdk/core/core-rest-pipeline/src/policies/setClientRequestIdPolicy.ts

```diff
@@ -9,20 +9,32 @@ import { PipelinePolicy } from "../pipeline";
  */
 export const setClientRequestIdPolicyName = "setClientRequestIdPolicy";
 
+/**
+ * Options for setting client request id details to outgoing requests.
+ */
+export interface SetClientRequestIdPolicyOptions {
+  /**
+   * If specified, the name of the header will be used to pass the request ID to.
+   * The default header name is `x-ms-client-request-id`.
+   */
+  requestIdHeaderName?: string;
+}
+
 /**
  * Each PipelineRequest gets a unique id upon creation.
  * This policy passes that unique id along via an HTTP header to enable better
  * telemetry and tracing.
- * @param requestIdHeaderName - The name of the header to pass the request ID to.
+ * @param options - Options that customize the policy.
  */
 export function setClientRequestIdPolicy(
-  requestIdHeaderName = "x-ms-client-request-id"
+  options: SetClientRequestIdPolicyOptions = {}
 ): PipelinePolicy {
   return {
     name: setClientRequestIdPolicyName,
     async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
-      if (!request.headers.has(requestIdHeaderName)) {
-        request.headers.set(requestIdHeaderName, request.requestId);
+      options.requestIdHeaderName = options.requestIdHeaderName ?? "x-ms-client-request-id";
```

Comment:
we should avoid mutating object parameters, since callers can re-use them (this caused a bug recently in Search)

prefer patterns that use a local variable like:

```ts
const requestIdHeaderName = options.requestIdHeaderName ?? "x-ms-client-request-id"
```

Reply:
Reverted back now.